src/main/java/org/fogbowcloud/saps/engine/core/scheduler/Scheduler.java

```diff
@@ -130,20 +157,24 @@ private void rollBackTaskState(SapsImage task) {
      * This function schedules up to tasks.
      */
     private void schedule() {
-        List<SapsImage> selectedTasks = selectTasks();
-        submitTasks(selectedTasks);
+        try {
+            List<SapsImage> selectedTasks = selectTasks();
+            submitTasks(selectedTasks);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
     }
 
     /**
      * This function selects tasks following a strategy for submit in Arrebol.
      *
      * @return selected tasks list
      */
-    protected List<SapsImage> selectTasks() {
+    List<SapsImage> selectTasks() throws Exception {
```

Comment:
Add some visible encapsulation to this method, `public` in this case.

Reply:
Done.